src/stdlib_experimental_default.f90

```diff
@@ -0,0 +1,142 @@
+module default_m
+  !!
+  !! Provides a generic function `default`, which can be used to
+  !! conveniently implement fallback values for optional arguments
+  !! to subprograms. If `x` is an `optional` parameter of a
+  !! subprogram, then the expression `default(x, y)` inside that
+  !! subprogram evaluates to `x` if it is present, otherwise `y`.
+  !!
+  !! It is an error to call `default` with a single actual argument.
+  !!
+  !! For additional clarity, `default` be called with keyword argument
+  !! for the fallback value, e.g., `default(x, to=1.0)`.
+  !!
+  use iso_fortran_env, only: sp => real32, dp => real64, qp => real128, int16, int32, int64
```

Comment:
At some points, it should use `stdlib_experimental_kinds` proposed in #63 . But I would say it is fine for now.